text/rfcs/0006-metadata-in-forc-manifest.md

```diff
@@ -0,0 +1,92 @@
+- Feature Name: Indexing Metadata Section in Forc Manifest
+- Start Date: 2024-01-25
+- Author(s): @deekerno
+- RFC PR: [FuelLabs/rfcs#21](https://github.com/FuelLabs/rfcs/pull/21)
+
+## Summary
+
+An optional section describing indexing-specific metadata should be added to the Forc manifest.
+
+## Motivation
+
+The objective of this change is to allow a Sway contract developer to describe the necessary information for indexing the contract in the Forc manifest, if desired. The expected outcome is that developers will be able to define indexing-specific _metadata_ in the same place and manner that they define other important information, e.g. authoring information, dependencies, etc. 
+
+## Guide-Level Explanation
+
+_Indexing-specific metadata_ refers to the set of information necessary to allow for an indexing service to begin processing resultant data from contract execution. The service should use this information to create distinct and separate tables for contract data as well as route creation on a publicly-available querying API.
+
+Consider the following example of a Forc project manifest: 
+```
+[project]
+authors = ["User"]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "my-fuel-project"
+
+[project.metadata.indexing]
+namespace = "examaple-namespace"
```

Comment:
I thought about this for awhile and I'm still not sure how to address this in the indexing metadata section.

For context, the namespace functionality was actually used as a way to separate indexers that may have the same name. Due to the original indexer's ability to be self-hosted, it was unlikely that namespaces would be collide since teams were generally expected to run their own indexer service deployment. Additionally, the original indexer also contained functionality to use `forc wallet` to place indexer uploading/stopping/removal behind authenticated routes.

The next iteration of the indexer is intended to be a sort of hosted "public good" service. Thus, the indexer team expects users to deploy their indexers to the service, and in order to ensure that database tables are distinct from one another (as they've generally been named after their GraphQL or Sway types), we need to determine a unique prefix for tables associated with the same contract. The prefix should be user-chosen as the developer experience with using a user-generated identifier is likely to be better than something generated that has no meaning to them. I did think about using a contract ID, but that will change if the content of the contract changes.

Reply:
Maybe the indexer website should provide the user with a unique key, which can be used as a prefix?

Are users paying for the indexing service? If so, it should be easy to have a key (or even user-chosen unique namespace) linked to their billing account.

Reply:
That's probably a good idea; users could request one using `forc wallet` and the service could handle generating a unique key in an idempotent manner.